andela/vidar-ah-backend
# #164139657 Implement in-app and email notifications

#### What does this PR do?
This PR implements in-app and email notifications the Authors Haven.

#### Description of Task to be completed?
Users should be able to *optionally* get an email and/or in-app notifications related to their activities on the app such as new comments, new followers, updated ratings and new likes. 

#### How should this be manually tested?
- Check .env.sample for the pusher environment variables needed to test the application
- Run `npm run db-migrate` to run migrations
- Run `npm run build` to build the application
- Run `npm start` to start the application
- Sign in with valid credentials
- Create a new article and have someone else comment on or like the article. You will get an email notification about the corresponding activities.
- You can catch up on missed notifications by hitting the endpoint `/api/v1/notifications` with a `GET` method.
- You can change your notification choice by hitting the end point `/api/v1/notifications` with a `PATCH` method providing one of four options in the body of the request in a variable named `notificationChoice`. The four available options are `both` (default), `email only`, ‘app only`, or `none`. 

#### Any background context you want to provide?
- To implement the notification on the front end you will need access to the API keys for the app, as well as the app channel and event which are listed as follows
Channel: `’vidar-notifications’`
Event: `’notify-[provide user id here without the square braces]’`

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139657

#### Screenshots (if appropriate)
- N/A

#### Questions:
N/A
